### ISSUE TITLE
Headless API does not include allowed file types for file upload field type

### ISSUE THREAD

**Issue by karlmacklin** [umbraco/Umbraco.Forms.Issues#923]
### Steps to reproduce

1. Set up a form with a field of type Upload, and restrict it to only a few specific allowed file types.
2. Fetch the form definition via the headless API.
3. Read the resulting definition JSON for the Upload field.

### Expected result

The field definition should include information about what field types are allowed.

### Actual result

The field definition is missing the file type restrictions.

**Comment by AndyButland** [comment]
Hi - this will be resolved in the next patch release.  We've updated the form definition made available to include this information on each field.

```
  "fileUploadOptions": {
      "allowAllUploadExtensions": false,
      "allowedUploadExtensions": [
          "png",
          "jpg",
          "gif",
          "svg"
      ],
      "allowMultipleFileUploads": true
  },
```

Just to note too I have seen your other issue raised about how exactly you can use file upload fields within the Forms API.  I'll need to spent a bit more time on that one as to be honest I hadn't considered that field type up to now, and I suspect there are changes we'll need to make server-side to support it.

**Comment by karlmacklin** [comment]
@AndyButland 
Yes I would expect so as well, as I have not seen a decent way to make a application/json request to dotnet apps that includes files, instead it's usually the multipart/form-data type of request with its boundary marking etc, a whole other beast.

Also, I should add regarding file types, we (development team) all found this UI **very** confusing:
![image](https://user-images.githubusercontent.com/353806/203966044-d5878b5f-596c-4f6d-b74e-5ed44ecad845.png)

We interpret that as we have turned OFF the enforcement of only allowing specific file types. Whereas toggling it on allows all file types instead.

Also adding a user defined file type is very confusing, as you add it to a list, but then you must refresh the whole page and reopen the editor for the upload field, and by then you can finally see it in the context of the rest of fields.

It's all workable, as long as you know the quirks I suppose, so take the feedback for what it is :)
Our client will probably find it more troublesome than we do.
